src/wallet/walletdb.cpp

```diff
@@ -1072,5 +1092,9 @@ std::unique_ptr<WalletDatabase> CreateDummyWalletDatabase()
 /** Return object for accessing temporary in-memory database. */
 std::unique_ptr<WalletDatabase> CreateMockWalletDatabase()
 {
+#ifdef USE_BDB
     return MakeUnique<BerkeleyDatabase>(std::make_shared<BerkeleyEnvironment>(), "");
+#elif USE_SQLITE
+    return MakeUnique<SQLiteDatabase>("", "", true);
```

Comment:
In commit "Do not compile BDB things when USE_BDB is defined" (8475409860d68e41e136129b59678fe1320cfb44)

It would be good to order commit "Fix mock SQLiteDatabases" (7b275da4608adc66ed6a7bbb1b807ca6dcb598e9) before this commit so mock databases aren't broken between commits by this call.

Also just a note in case other reviewers were curious like me but an `""` database filename will create a unnamed temporary database (https://www.sqlite.org/inmemorydb.html). `":memory:"` could also be used to create a database guaranteed to be in memory and never written to disk, but `""` seems most appropriate here

Reply:
I guess so, but tests with BDB disabled are still broken after this PR so I didn't feel the need to do that. Arguable that change really belongs in #20267.